### ISSUE TITLE
To JSON conversion: Delete causes exception

### ISSUE THREAD

**Issue by Tamriel** [fthomas/crjdt#19]
```scala
import eu.timepit.crjdt.circe.RegNodeConflictResolver.LWW
import eu.timepit.crjdt.circe.syntax._
import eu.timepit.crjdt.core.Replica
import eu.timepit.crjdt.core.syntax._

object Playground {

  def main(args: Array[String]): Unit = {
    var p = Replica.empty("p")
    val list = doc.downField("list")
    p = p.applyCmd(list := `[]`)
    p = p.applyCmd(list.iter.insert("1"))
    p = p.applyCmd(list.iter.next.insert("2"))
    p = p.applyCmd(list.iter.next.delete) // it works without this line
    println(p.document.toJson)
  }
}
```
@TanUkkii007: The above code causes
```
Exception in thread "main" scala.MatchError: (RegT(IdK(Id(2,p))),RegNode(Map())) (of class scala.Tuple2)
	at eu.timepit.crjdt.circe.NodeToJson$$anonfun$listToJson$1.apply(NodeToJson.scala:50)
```

**Comment by TanUkkii007** [comment]
Thanks for reporting!
I will check it in the weekend.

**Comment by TanUkkii007** [comment]
> (RegT(IdK(Id(2,p))),RegNode(Map()))

It seems the document state has empty register. Could be related to #15 .

**Comment by TanUkkii007** [comment]
Will be fixed on #20 .
The reason of the MatchError is wrong usage of `foreach` with PartialFunction filtering items that are not present. 
No relation to #15 .
